File: patterns/java/src/refactoring_guru/patterns/visitor/example/shapes/Dot.java
package refactoring_guru.patterns.visitor.example.shapes;

import refactoring_guru.patterns.visitor.example.visitor.Visitor;

public class Dot implements Shape {
    public int id;
    public int x;
    public int y;

    public Dot() {
    }

    public Dot(int id, int x, int y) {
        this.id = id;
        this.x = x;
        this.y = y;
    }

    @Override
    public void move(int x, int y) {
        // move shape
    }

    @Override
    public void draw() {
        // draw shape
    }

    public String accept(Visitor visitor) {
        return visitor.visitDot(this);
    }

    public int getX() {
        return x;
    }

    public int getY() {
        return y;
    }

    public int getId() {
        return id;
    }
}

File: patterns/cs/src/RefactoringGuru/Visitor/Example/Shapes/Dot.cs
using System;
using RefactoingGuru.Visitor.Example.Visitor;

namespace RefactoingGuru.Visitor.Example.Shapes
{
    class Dot : Shape
    {
        public int id;
        public int x;
        public int y;

        public Dot() {}

        public Dot(int id, int x, int y)
        {
            this.id = id;
            this.x = x;
            this.y = y;
        }

        public void Move(int x, int y)
        {
            // move shape
        }

        public void Draw()
        {
            // draw shape
        }

        public String Accept(IVisitor visitor)
        {
            return visitor.VisitDot(this);
        }

        public int GetX()
        {
            return x;
        }

        public int GetY()
        {
            return y;
        }

        public int GetId()
        {
            return id;
        }
    }
}
